pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -726,6 +726,11 @@ impl Timeline {
                     .await?;
 
                 self.upload_new_image_layers(image_layers)?;
+                if !is_complete {
+                    // Yield and do not do any other kind of compaction.
+                    info!("skipping shard ancestor compaction due to pending image layer generation tasks (preempted by L0 compaction).");
```

Comment:
While we're here, can we add some logging with stats about image layer compaction regardless of the outcome? E.g. how long it ran for, number of layers/bytes processed, stuff like that.

Reply:
I added it at the end of `create_image_layers` https://github.com/neondatabase/neon/pull/10572/commits/5f03e9d933a94a8d86b555d3385c4d919416a03f